### PR TITLE
Prefix custom attributes with instrumentation name

### PR DESCRIPTION
**Pull request by trask**
Related to #602

I haven't updated the tests yet, will do that if/when there's agreement on this change.

### CODE REVIEW

File: instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/TracingExecutionInterceptor.java [hunk 1]
```diff
@@ -137,9 +137,9 @@ private void decorateWithExAttributesData(Span span, ExecutionAttributes attribu
     String awsServiceName = attributes.getAttribute(SdkExecutionAttribute.SERVICE_NAME);
     String awsOperation = attributes.getAttribute(SdkExecutionAttribute.OPERATION_NAME);
 
-    span.setAttribute("aws.agent", COMPONENT_NAME);
-    span.setAttribute("aws.service", awsServiceName);
-    span.setAttribute("aws.operation", awsOperation);
+    span.setAttribute("aws-sdk.agent", COMPONENT_NAME);
```
Comment (anuraaga) [inline instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/TracingExecutionInterceptor.java:140]: The aws changes will be very breaking for us in the collector. I need to get back to my spec PR for these... Do intend to change these, but need to change the collector handling in a backwards compatible way first.

Comment (trask) [reply instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/TracingExecutionInterceptor.java]: I reverted the aws attribute name changes

**Review by anuraaga — APPROVED** [review]

**Review by iNikem — APPROVED** [review]

File: instrumentation/apache-camel-2.20/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/decorators/KafkaSpanDecorator.java [hunk 2]
```diff
@@ -61,22 +61,22 @@ public void pre(Span span, Exchange exchange, Endpoint endpoint, CamelDirection
 
     String partition = getValue(exchange, PARTITION, Integer.class);
     if (partition != null) {
-      span.setAttribute("partition", partition);
+      span.setAttribute("apache-camel.kafka.partition", partition);
```
Comment (mateuszrzeszutek) [inline instrumentation/apache-camel-2.20/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/decorators/KafkaSpanDecorator.java:64]: What do you think about using `SemanticAttributes.MESSAGING_KAFKA_PARTITION` for this one?

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by iNikem** [comment]
@trask care to wrap it up?

**Pull request merged**